File: Decorator/Java/WeaponSample/src/Scope.java
public class Scope extends WeaponAttachment {
    public Scope(IWeapon weapon) {
        super(weapon);
        this.name = "Scope X8";
    }

    @Override
    public void Fire() {
        weapon.Fire();
        System.out.println("    " + this.name);
    }
}

File: Decorator/csharp/Sample/Decorator/Weapon/Scope.cs
using System;

namespace Decorator.Weapon
{
    public class Scope  : WeaponAttachment
    {
        public Scope(IWeapon weapon) : base(weapon)
        {
        }

        public override void Fire()
        {
            _weapon.Fire();
            Console.WriteLine("     Scope x4");
        }

        
    }
}